hoomd/write/__init__.py

```diff
@@ -30,3 +33,7 @@
 from hoomd.write.gsd_burst import Burst
 from hoomd.write.dcd import DCD
 from hoomd.write.table import Table
+try:
```

Comment:
Move the try/except around `import h5py` in `write/hdf5.py`. Will also need to add an error check (e.g. if `h5py is None`) check in `HDF5Logger`. This way, HOOMD developers that modify `write/hd5.py` will see the cause of any `ImportError` related to `hdf5.py` and the `HDF5Logger` class is available (e.g. for sphinx doc builds) when `h5py` is not installed.